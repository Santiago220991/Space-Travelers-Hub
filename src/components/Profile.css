.reserved-lists {
  display: flex;
  align-items: flex-start;
  margin-left: 5%;
  margin-right: 5%;
}

.rockets-reserved-list {
  border: solid 1px rgb(179, 174, 174);
  border-bottom: none;
  border-radius: 2px;
}

.rockets-list-title {
  margin-top: 0;
  font-size: 24px;
  font-weight: bold;
}

.rockets-reserved-list-container {
  margin-left: 3%;
  width: 40%;
}

.rocket-reserved {
  border-bottom: solid 1px rgb(179, 174, 174);
  height: 30px;
}
