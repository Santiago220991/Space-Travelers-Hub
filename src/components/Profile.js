import React from 'react';

const Profile = () => (
  <div>Hola</div>
);

export default Profile;
