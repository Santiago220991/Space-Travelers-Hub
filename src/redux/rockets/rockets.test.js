import React from 'react';
import renderer from 'react-test-renderer';
import RocketList from '../../components/__mocks__/RocketCard';
import '@testing-library/jest-dom';
import Profile from '../../components/Profile'
import { Provider } from 'react-redux';
import Store from '../configureStore';
import { render } from '@testing-library/react';
import RocketReducer from './rockets';



test('Render Profile', () => {
  const tree = renderer.create(<Provider store={Store}><Profile /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});

describe("Rockets Page Test", ()=>{
test('Render Rocket Pages', () => {
    const tree = renderer.create(<Provider store={Store}><RocketList /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('There should be "The Falcon 1" as a rocket name', () => {
    const {container}  = render(<Provider store={Store}><RocketList /></Provider>)
    expect(container).toHaveTextContent('The Falcon 1');
  });
  test('There is be a "Reserved" rocket', () => {
    const {container}  = render(<Provider store={Store}><RocketList /></Provider>)
    expect(container).toHaveTextContent('Reserved');
  });
  test('There is be a "Reserved" rocket with a "Cancel Reservation" button', () => {
    const {container}  = render(<Provider store={Store}><RocketList /></Provider>)
    expect(container).toHaveTextContent('Reserved');
    const cancelbtn = container.querySelector('.unreserve-rocket');
    expect(cancelbtn).toHaveTextContent('Cancel Reservation');
  });
})

describe("Rockets Reducer Tests", ()=>{
    test('Test Get Rocket List',()=>{
        const GET_ROCKET="GET_ROCKET"
        const payload=[
            {
                id: 'Falcon 1',
                flickr_images: 'https://imgur.com/DaCfMsj.jpg',
                info: 'The Falcon 1 was an expendable launch system privately.',
                name: 'Falcon 1',
              },
              {
                id: 'Falcon 9',
                flickr_images: 'https://imgur.com/DaCfMsj.jpg',
                info: 'The Falcon 1 was an expendable launch system privately.',
                name: 'Falcon 1',
                
              },
        ]
        const action = { type: GET_ROCKET, payload };
        const result=RocketReducer([],action)
        expect(result).toEqual(payload)
    })
    test('Test Reserve a Rocket',()=>{
        const RESERVE='RESERVE';
        const action={type: RESERVE, payload: 'Falcon 1'}
        const state=[
            {
                id: 'Falcon 1',
                flickr_images: 'https://imgur.com/DaCfMsj.jpg',
                info: 'The Falcon 1 was an expendable launch system privately.',
                name: 'Falcon 1',
              },
              {
                id: 'Falcon 9',
                flickr_images: 'https://imgur.com/DaCfMsj.jpg',
                info: 'The Falcon 1 was an expendable launch system privately.',
                name: 'Falcon 1',
                
              },
        ]
        const newState=[
            {
                id: 'Falcon 1',
                flickr_images: 'https://imgur.com/DaCfMsj.jpg',
                info: 'The Falcon 1 was an expendable launch system privately.',
                name: 'Falcon 1',
                reserved: true,
              },
              {
                id: 'Falcon 9',
                flickr_images: 'https://imgur.com/DaCfMsj.jpg',
                info: 'The Falcon 1 was an expendable launch system privately.',
                name: 'Falcon 1',
                
              },
        ]
        const result=RocketReducer(state,action)
        expect(result).toEqual(newState)
    })
    test('Test Cancel Reservation',()=>{
        const UNRESERVE='UNRESERVE';
        const action= {type: UNRESERVE, payload: 'Falcon 1'}
        const state=[
            {
                id: 'Falcon 1',
                flickr_images: 'https://imgur.com/DaCfMsj.jpg',
                info: 'The Falcon 1 was an expendable launch system privately.',
                name: 'Falcon 1',
                reserved: true,
              },
              {
                id: 'Falcon 9',
                flickr_images: 'https://imgur.com/DaCfMsj.jpg',
                info: 'The Falcon 1 was an expendable launch system privately.',
                name: 'Falcon 1',
                
              },
        ]
        const newState=[
            {
                id: 'Falcon 1',
                flickr_images: 'https://imgur.com/DaCfMsj.jpg',
                info: 'The Falcon 1 was an expendable launch system privately.',
                name: 'Falcon 1',
                reserved: false,
              },
              {
                id: 'Falcon 9',
                flickr_images: 'https://imgur.com/DaCfMsj.jpg',
                info: 'The Falcon 1 was an expendable launch system privately.',
                name: 'Falcon 1',
                
              },
        ]
        const result=RocketReducer(state,action)
        expect(result).toEqual(newState)
    })
})