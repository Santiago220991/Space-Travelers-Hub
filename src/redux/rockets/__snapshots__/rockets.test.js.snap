// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Profile 1`] = `
<div
  className="reserved-lists"
>
  <div
    className="rockets-reserved-list-container"
  >
    <p
      className="rockets-list-title"
    >
      My Missions
    </p>
    <div
      className="rockets-reserved-list"
    />
  </div>
  <div
    className="rockets-reserved-list-container"
  >
    <p
      className="rockets-list-title"
    >
      My Rockets
    </p>
    <div
      className="rockets-reserved-list"
    />
  </div>
</div>
`;

exports[`Rockets Page Test Render Rocket Pages 1`] = `
Array [
  <div
    className="rocket-card"
    id="Falcon 1"
  >
    <img
      alt="rocket"
      className="img-rocket"
      src="https://imgur.com/DaCfMsj.jpg"
    />
    <div
      className="rocket-div"
    >
      <p
        className="name-rocket"
      >
        Falcon 1
      </p>
      <p
        className="info-rocket"
      >
        <span
          className="reserved-badge"
        >
          Reserved
        </span>
        The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.
      </p>
      <button
        className="unreserve-rocket"
        type="button"
      >
        Cancel Reservation
      </button>
    </div>
  </div>,
  <div
    className="rocket-card"
    id="Falcon 9"
  >
    <img
      alt="rocket"
      className="img-rocket"
      src="https://imgur.com/DaCfMsj.jpg"
    />
    <div
      className="rocket-div"
    >
      <p
        className="name-rocket"
      >
        Falcon 1
      </p>
      <p
        className="info-rocket"
      >
        The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.
      </p>
      <button
        className="reserve-rocket"
        type="button"
      >
        Reserve Rocket
      </button>
    </div>
  </div>,
  <div
    className="rocket-card"
    id="Falcon 2"
  >
    <img
      alt="rocket"
      className="img-rocket"
      src="https://imgur.com/DaCfMsj.jpg"
    />
    <div
      className="rocket-div"
    >
      <p
        className="name-rocket"
      >
        Falcon 1
      </p>
      <p
        className="info-rocket"
      >
        <span
          className="reserved-badge"
        >
          Reserved
        </span>
        The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.
      </p>
      <button
        className="unreserve-rocket"
        type="button"
      >
        Cancel Reservation
      </button>
    </div>
  </div>,
]
`;
